brouter-core/src/main/java/btools/router/RoutingContext.java

```diff
@@ -404,28 +402,23 @@ else if ( s1 > s2 )
     return (int)(d + 1.0 );
   }
 
-  // assumes that calcDistance/calcCosAngle called in sequence, so coslat valid
   public double getCosAngle()
   {
     return cosangle;
   }
 
-  public double getCosLat()
-  {
-    return coslat;
-  }
-
   public double calcAngle( int lon0, int lat0,  int lon1, int lat1, int lon2, int lat2 )
   {
-    double dlat1 = (lat1 - lat0);
-    double dlon1 = (lon1 - lon0) * coslat;
-    double dlat2 = (lat2 - lat1);
-    double dlon2 = (lon2 - lon1) * coslat;
+    double[] lonlat2m = CheapRulerSingleton.getLonLatToMeterScales( lat1 );

```

Comment:
We can use kx and ky here to compute the direction without relying on coslat, I think.